foreshadow/tests/test_transformers/test_smart/test_smart.py

```diff
@@ -3,6 +3,86 @@
 from foreshadow.utils.testing import get_file_path
 
 
+@pytest.fixture()
+def smart_child():
+    """Get a defined SmartTransformer subclass, TestSmartTransformer.
+
+    Note:
+        Always returns StandardScaler.
+
+    """
+    from foreshadow.smart import SmartTransformer
+    from foreshadow.concrete import StandardScaler
+
+    class TestSmartTransformer(SmartTransformer):
+        def pick_transformer(self, X, y=None, **fit_params):
+            return StandardScaler()
+
+    yield TestSmartTransformer
+
+
+@pytest.mark.parametrize("deep", [True, False])
+def test_smart_get_params_default(smart_child, deep):
+    """Ensure that default get_params works.
+
+    Args:
+        smart_child: a smart instance
+        deep: deep param to get_params
+
+    """
+    smart = smart_child()
+    params = smart.get_params(deep=deep)
+    default_state = {
+        "check_wrapped": True,
+        "column_sharer": None,
+        "force_reresolve": False,
+        "keep_columns": False,
+        "name": None,
+        "should_resolve": True,
+        "transformer": None,
+        "y_var": False,
+    }
+    assert default_state == params
+
+
+@pytest.mark.parametrize(
+    "initial_transformer", [None, "BoxCox", "StandardScaler"]
```

Comment:
Not sure I understand the test case here. If the `smart.transformer` after `set_params` is always `StandardScaler`, then what's the point of the first two cases for None and BoxCox?